### PR TITLE
Fixed Duration and added Async Methods

### PR DESCRIPTION
**Pull request by mrousavy**
* Fixed Toast Show Duration and Message sometimes not changing
* Added new Async Methods which are awaitable to the point until the Toast disappears
__Why is it called **Toasty** btw? It's an Android Snackbar... [That's a toast ;)](https://github.com/mrousavy/Toast)__

### CODE REVIEW

File: Toast/Toasty.xaml.cs [hunk 1]
```diff
@@ -1,233 +1,240 @@
...
-		private static void OnMessageChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
-		{
-			Toasty toast = d as Toasty;
-			if (toast == null)
-			{
-				return;
-			}
+            if(!string.IsNullOrWhiteSpace(Message)) {
+                TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
+                EventHandler onComplete = null;
```
Comment (bengry) [inline Toast/Toasty.xaml.cs]: Why split the deceleration and implementation?

**Review by bengry — APPROVED** [review]
Looks good overall. I added a small question which I would appreciate a comment on, but otherwise - feel free to merge.

File: Toast/Toasty.xaml.cs (hunk 1 above)
Comment (mrousavy) [inline Toast/Toasty.xaml.cs]: I have to remove the Event once completed, that doesn't work if I initialize it directly.
I cannot reference `onComplete` when I'm inside the inner Method;
`var onComplete = (s,e) => { /*Not possible to reference onComplete here*/ }`

Comment (mrousavy) [inline Toast/Toasty.xaml.cs]: Btw, can you update it on NuGet @bengry ? Thanks!

**Pull request merged**